desktop/core/src/desktop/lib/raz/raz_client.py

```diff
@@ -144,17 +142,47 @@ def check_access(self, method, url, params=None, headers=None, data=None):
   def _handle_raz_req(self, raz_url, request_headers, request_data):
     if self.auth_type == 'kerberos':
       auth_handler = requests_kerberos.HTTPKerberosAuth(mutual_authentication=requests_kerberos.OPTIONAL)
-      raz_req = requests.post(raz_url, headers=request_headers, json=request_data, auth=auth_handler, verify=False)
+      raz_response = self._handle_raz_ha(raz_url, headers=request_headers, data=request_data, auth_handler=auth_handler)
+
     elif self.auth_type == 'jwt':
       jwt_token = fetch_jwt()
-
       if jwt_token is None:
         raise PopupException('Knox JWT is not available to send to RAZ.')
 
       request_headers['Authorization'] = 'Bearer %s' % (jwt_token)
-      raz_req = requests.post(raz_url, headers=request_headers, json=request_data, verify=False)
+      raz_response = self._handle_raz_ha(raz_url, headers=request_headers, data=request_data)
+
+    if not raz_response:
+      raise PopupException('Failed to receive a valid response from RAZ.')
+
+    return raz_response
+
+
+  def _handle_raz_ha(self, raz_url, headers, data, auth_handler=None, verify=False):
+    raz_urls_list = raz_url.split(',')
+    params = {
+      'headers': headers, 
+      'json': data,
+      'verify': verify
+    }
+
+    if auth_handler:
+      params['auth'] = auth_handler
+
+    raz_response = None
+    for r_url in raz_urls_list:
+      r_url = "%s/api/authz/%s/access?doAs=%s" % (r_url.rstrip('/'), self.service, self.username)
+      LOG.info('Attempting to connect to RAZ URL: %s' % r_url)
+
+      try:
+        raz_response = requests.post(r_url, **params)
+      except Exception as e:
+        if 'Failed to establish a new connection' in str(e):
+          LOG.warning('Raz URL %s is not available.' % r_url)
 
-    return raz_req
+      # Check response for None and if response code is successful (200), return the raz response.
+      if (raz_response is not None) and (raz_response.status_code == 200):
```

Comment:
Does the Exception and this condition cover all possible scenarios? Or can the status code be something else than 200 here and if so, what happens then? (I don't know this API but in general there are a range of successful response and redirect status codes)